### PR TITLE
Feat/actions

### PR DESCRIPTION
**Pull request by wildintellect**
Automatically build and push the JupyterBook everytime a merge is made to `main`
Solves #2

### CODE REVIEW

**Comment by wildintellect** [comment]
@lillythomas I believe I've done this correctly but going to ask @chuckwondo to review before merging. Once in place then all future PR merges to `main` will rebuild the website automatically.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+jobs:
+  deploy-book:
+    runs-on: ubuntu-latest
+    # Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+    permissions:
+      pages: write
+      id-token: write
+    steps:
+      - uses: actions/checkout@v3
```
Comment (chuckwondo) [inline .github/workflows/deploy.yml]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+      # Install dependencies
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
```
Comment (chuckwondo) [inline .github/workflows/deploy.yml]:
```suggestion
        uses: actions/setup-python@v5
```

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        with:
+          python-version: 3.11
```
Comment (chuckwondo) [inline .github/workflows/deploy.yml:32]:
```suggestion
          python-version: 3.11
          cache: 'pip' # cache pip dependencies
```

File: book/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+geopandas==0.10.2
+matplotlib==3.5.1
+numpy==1.21.2
+pandas==1.3.5
```
Comment (chuckwondo) [inline book/requirements.txt:4]: `pandas` v2 was released a year ago, so perhaps we should bump this? Are there other dependencies here that don't play nicely with v2?

File: book/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+plotly==5.3.1
+pystac-client==1.1.0
+requests==2.26.0
+rasterio==1.2.10
+scikit-image==0.18.3
+scikit-learn==0.24.2
+stackstac==0.4.0
+xarray==0.19.0
```
Comment (chuckwondo) [inline book/requirements.txt:12]: `xarray` graduated to releases based on the current year over 2 years ago, and `0.19.0` is nearly 3 years old, so I suggest we bump this to the latest. In general, I suggest we bump as many of these dependencies to the latest possible (whatever can be fully resolved in conjunction).

**Review by chuckwondo — CHANGES_REQUESTED** [review]
Nice work on incorporating GitHub Actions. I'd just like to see `pip` caching added, along with bumping as many things (actions and packages) to the latest versions that work together (some are rather old).

**Comment by wildintellect** [comment]
@chuckwondo thanks I only caught a few of the outdated actions (this was made using jupyterbook cookiecutter)
As for the packages in the moved requirements file (made no changes), that should be a separate issue we bring up with @lillythomas - there's also an open question about switching it `conda`
The only requirement related to the building is in it's own file `jupyter-book`

File: book/requirements.txt (hunk 4 above)
Comment (wildintellect) [reply book/requirements.txt:4]: this file is not part of the github action

File: book/requirements.txt (hunk 5 above)
Comment (wildintellect) [reply book/requirements.txt:12]: this file is not part of the github action

**Review by chuckwondo — APPROVED** [review]

**Pull request merged**